### ISSUE TITLE
`MaskedSquaredErrorMetric` doesn't match Keras' Built-in `MeanSquaredErrorMetric` for Unit Mask

### ISSUE THREAD

**Issue by jeipollack** [CosmoStat/wf-psf#170]
While labelled a `bug` this is not a bug.  It is just a question of understanding.

We introduced custom loss and metrics classes for mask handling.   We ran a simple sanity test by providing a unitary mask to test against the unmasked case which would use Keras' built-in `MeanSquaredErrorMetric`. However, we monitor "loss" if the mask is provided versus when it is not where we monitor the `"mean_squared_error"`.  Here is the conditional for both cases:

```
 if training_handler.training_hparams.loss == "mask_mse":
        loss = train_utils.MaskedMeanSquaredError()
        monitor = "loss"
        param_metrics = [train_utils.MaskedMeanSquaredErrorMetric()]
        non_param_metrics = [train_utils.MaskedMeanSquaredErrorMetric()]
        outputs = tf.stack(
            [data_conf.training_data.dataset["noisy_stars"], data_conf.training_data.dataset["masks"]], axis=-1
        )
        output_val = tf.stack(
            [data_conf.test_data.dataset["stars"], data_conf.test_data.dataset["masks"]], axis=-1
        )
    else:
        loss = tf.keras.losses.MeanSquaredError()
        monitor = "mean_squared_error"
        param_metrics = [tf.keras.metrics.MeanSquaredError()]
        non_param_metrics = [tf.keras.metrics.MeanSquaredError()]
        outputs = data_conf.training_data.dataset["noisy_stars"]
        output_val = data_conf.test_data.dataset["stars"]
```

**Expectation:**

Comparing both cases with a unit mask I would have expected the losses, mean_squared_errors, val_loss, and val_mean_squared_errors to be equivalent.

But they are not:

```
Epoch 1: loss improved from inf to 0.00001, saving model to /gpfswork/rech/ynx/uuu68hq/validation/c
ase_study/vxx/v0xx/v0_2_0/wf-outputs/wf-outputs-202507231415/checkpoint/checkpoint_callback_polywf_
simulated_dataset_v0_2_0_cycle1
63/63 - 152s - loss: 6.0427e-06 - masked_mean_squared_error: 5.9251e-06 - val_loss: 1.6639e-06 - va
l_masked_mean_squared_error: 1.6547e-06 - 152s/epoch - 2s/step


Epoch 1: mean_squared_error improved from inf to 0.00001, saving model to /gpfswork/rech/ynx/uuu68hq/validation/gh-165/vxx/v0xx/v0_1_0/wf
-outputs/wf-outputs-202507241156/checkpoint/checkpoint_callback_polywf_simulated_dataset_v0_1_0_cycle1
63/63 - 157s - loss: 6.4974e-06 - mean_squared_error: 6.2680e-06 - val_loss: 1.4982e-06 - val_mean_squared_error: 1.4982e-06 - 157s/epoch
 - 2s/step
```

It just so happened though that they converged at the same point:

```
Epoch 47: loss improved from 0.00000 to 0.00000, saving model to /gpfswork/rech/ynx/uuu68hq/validat
ion/case_study/vxx/v0xx/v0_2_0/wf-outputs/wf-outputs-202507231415/checkpoint/checkpoint_callback_po
lywf_simulated_dataset_v0_2_0_cycle2
63/63 - 139s - loss: 9.7586e-07 - masked_mean_squared_error: 1.3996e-06 - val_loss: 4.8626e-09 - va
l_masked_mean_squared_error: 4.8324e-09 - 139s/epoch - 2s/step

Epoch 47: loss improved from 0.00000 to 0.00000, saving model to /gpfswork/rech/ynx/uuu68hq/validation/case_study/vxx/v0xx/v0_2_0/wf-outp
uts/wf-outputs-202507231415/checkpoint/checkpoint_callback_polywf_simulated_dataset_v0_2_0_cycle2
63/63 - 139s - loss: 9.7586e-07 - masked_mean_squared_error: 1.3996e-06 - val_loss: 4.8626e-09 - val_masked_mean_squared_error: 4.8324e-0
9 - 139s/epoch - 2s/step
Epoch 48/50
```

It doesn't seem to be a coincidence because I tested with two runs of 2 cycles of training where the different learning rates and number of epochs were used for the two runs and yet the models converged at the same point.

Run 2 showing where the two built-in and custom loss and metrics converged at the same epoch.

```
Epoch 116: loss improved from 0.00000 to 0.00000, saving model to /gpfswork/rech/ynx/uuu68hq/valida
tion/case_study/vxx/v0xx/v0_2_0/wf-outputs/wf-outputs-202507231340/checkpoint/checkpoint_callback_p
olywf_simulated_dataset_v0_2_0_cycle2
63/63 - 150s - loss: 9.7640e-07 - masked_mean_squared_error: 1.4062e-06 - val_loss: 8.3219e-09 - va
l_masked_mean_squared_error: 8.2378e-09 - 150s/epoch - 2s/step

Epoch 116: loss improved from 0.00000 to 0.00000, saving model to /gpfswork/rech/ynx/uuu68hq/validation/case_study/vxx/v0xx/v0_2_0/wf-out
puts/wf-outputs-202507231340/checkpoint/checkpoint_callback_polywf_simulated_dataset_v0_2_0_cycle2
63/63 - 150s - loss: 9.7640e-07 - masked_mean_squared_error: 1.4062e-06 - val_loss: 8.3219e-09 - val_masked_mean_squared_error: 8.2378e-0
9 - 150s/epoch - 2s/step
```

**Comment by jeipollack** [comment]
I was looking at Keras' built-in `MeanSquaredError` metric and it averages per pixel not batch as our custom metric does.
I am going to investigate in a test branch changing the custom `MaskedMeanSquaredErrorMetric` to match what Keras does.
